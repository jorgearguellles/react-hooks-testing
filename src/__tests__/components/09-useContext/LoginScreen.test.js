import React from "react";
import '@testing-library/jest-dom';
import { mount } from "enzyme";
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Testing <LoginScreen />', ()=>{

  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={ { setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('1. SnapShot', ()=>{
    expect( wrapper ).toMatchSnapshot();
  });

  test('2. Should call setUser with right arguments', ()=>{
    wrapper.find('button').prop('onClick')();
    expect( setUser ).toBeCalledWith({
      id: 123,
      name: 'Fernando'
    });
  })
})