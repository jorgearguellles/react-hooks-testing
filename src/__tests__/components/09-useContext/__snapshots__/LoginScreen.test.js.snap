// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <LoginScreen /> 1. SnapShot 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
